File: Netty/netty08-thrift/src/main/java/com/ayy/client/ThriftClient.java
package com.ayy.client;

import com.ayy.thrift.generated.Person;
import com.ayy.thrift.generated.PersonService;
import org.apache.thrift.protocol.TCompactProtocol;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.transport.TFramedTransport;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;

/**
 * @ Description
 * @ Author Zhao JIN
 * @ Date 13/05/2021
 * @ Version 1.0
 */

public class ThriftClient {
    public static void main(String[] args) {
        TTransport transport = new TFramedTransport(new TSocket("localhost", 10000));
        TProtocol protocol = new TCompactProtocol(transport);

        PersonService.Client client = new PersonService.Client(protocol);

        try {
            transport.open();

            Person person = client.getPersonByUsername("me");

            System.out.println(person.getUsername()+"--"+person.getAge()+"--"+person.isMarried());
            System.out.println(person);

            System.out.println("------");

            Person person1 = new Person();
            person1.setUsername("me2");
            person1.setAge(18);
            person1.setMarried(false);

            client.savePerson(person1);
        } catch (Exception e){
            throw new RuntimeException(e.getMessage(), e);
        } finally {
            transport.close();
        }
    }
}

File: Netty/netty08-thrift/py/com/ayy/thrift/py/client/ThriftClient.py
# _*_ coding:utf-8 _*_
__author__ = "Zhao JIN"

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import  TCompactProtocol
from com.ayy.thrift.py.generated import PersonService
from com.ayy.thrift.py.generated.ttypes import Person

transport = None
try:
    tSocket = TSocket.TSocket('localhost', 10000)
    tSocket.setTimeout(900)

    transport = TTransport.TFramedTransport(tSocket)
    protocol = TCompactProtocol.TCompactProtocol(transport)

    client = PersonService.Client(protocol)
    transport.open()

    person = client.getPersonByUsername("Me")
    print(person)
    print(person.username)
    print(person.age)
    print(person.married)

    person2 = Person()
    person2.username = "Others"
    person2.age = 18
    person2.married = False

    client.savePerson(person2)
except Thrift.TException as tx:
    print('%s' % tx.message)
finally:
    transport.close()
